core/src/main/java/com/expedia/www/haystack/client/Span.java

```diff
@@ -88,6 +88,7 @@ protected synchronized void finishTrace(long finishMicros) {
         this.endTime.compareAndSet(null, finishMicros);
         this.duration.compareAndSet(null, endTime.get() - startTime.get());
         finished.compareAndSet(false, true);
```

Comment:
What should happen if finished.compareAndSet(false, true) fails because the value is already true? It just doesn't get set to true, but it's already true, right? If so, finished.set(true) would work just as well.